pkg/library/libraryapplyconfiguration/apply_configuration.go

```diff
@@ -74,7 +74,30 @@ func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutat
 		errs = append(errs, fmt.Errorf("%d output-resource were produced, but not present in the specified output: %v", len(unspecifiedOutputIdentifiers), strings.Join(unspecifiedOutputIdentifiers, ", ")))
 	}
 
-	return errors.Join(errs...)
+	return errs
+}
+
+func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutationsGetter, allAllowedOutputResources *libraryoutputresources.OutputResources) []error {
+	errs := []error{}
+
+	if allDesiredMutationsGetter == nil {
+		return []error{fmt.Errorf("applyConfiguration is required")}
+	}
+
+	allMutationRequests := []manifestclient.SerializedRequestish{}
+	for _, clusterType := range sets.List(AllClusterTypes) {
+		desiredMutationsGetter := allDesiredMutationsGetter.MutationsForClusterType(clusterType)
+		switch {
+		case desiredMutationsGetter == nil:
+			errs = append(errs, fmt.Errorf("mutations for %q are required even if empty", clusterType))
+		case desiredMutationsGetter.GetClusterType() != clusterType:
+			errs = append(errs, fmt.Errorf("mutations for %q reported type=%q", clusterType, desiredMutationsGetter.GetClusterType()))
+		}
+
+		allMutationRequests = append(allMutationRequests, desiredMutationsGetter.Requests().AllRequests()...)
```

Comment:
It looks like we're just appending to this slice, but not using it for anything else. Can it be dropped?

Reply:
This function duplicates logic from `ValidateUnfilteredMutations`, right? We could extract the shared code into a helper and use it in both places.